docs/design/free_credit_offer_typesDesignDoc.md

```diff
@@ -0,0 +1,253 @@
+  Users() Users
+  // Credits is a getter for Credits repository
+  Credits() Credits
+	// Projects is a getter for Projects repository
+	Projects() Projects
+	// ProjectMembers is a getter for ProjectMembers repository
+	ProjectMembers() ProjectMembers
+	// APIKeys is a getter for APIKeys repository
+	APIKeys() APIKeys
+	// BucketUsage is a getter for accounting.BucketUsage repository
+	BucketUsage() accounting.BucketUsage
+	// RegistrationTokens is a getter for RegistrationTokens repository
+	RegistrationTokens() RegistrationTokens
+	// UsageRollups is a getter for UsageRollups repository
+	UsageRollups() UsageRollups
+
+	// BeginTransaction is a method for opening transaction
+  BeginTx(ctx context.Context) (DBTx, error)
+}
+```
+
+**satellite/offer/offerweb/server.go**
+- Open a new private port on the satellite for admin users to manage referral offer configuration
+
+```golang
+// NewServer creates a new instance of offerweb server
+Func NewServer(logger *zap.Logger, config Config, service *offer.Service, listener net.Listener) *Server {}
+
+// Register offerweb server onto satellite
+peer.Offer.Endpoint = offerweb.NewServer(logger, config, service, listener)
+```
+
+### Referral Links
+
+The referral link url will be a static url that contains userid as the unique identifier
+Exp: `https://mars.tardigrade.io/?uuid=<userid>`
+
+### Segment.io service
+
+**pkg/analytics/analytics.go**
+
+- For email service, we will design several event triggers in customer.io using their event triggered campaign. We will be using analytics-go package for back-end server and analytics.js for Satellite GUI from segment.io to send our trigger event to customer.io We will create a new package in storj/pkg for analytics that will check DNT first before sending data to customer.io
+- We will add a new configuration for storing segment tracking Id specifically for tardigrade branded satellite
+
+```golang
+client := analytics.New()
+```
+
+How to send an event from satellite to customer.io?
+
+```golang
+client.Enqueue(analytics.Track{
+  UserId: "f4ca124298",
+  Event:  "sign-ups",
+  Properties: analytics.NewProperties().Set("referred", "true"),
+})
+```
+
+How to send an event from Satellite GUI to customer.io?
+
+```javascript
+analytics.track('Signed Up', {
+  referred_by: ‘brandonisawesome’
+});
+```
+
+**List of trigger events**:
+- Send_referral event - triggered from the Satellite GUI
+- Referral_redeemed event - triggered from the consoleweb server.
+Reason: activation success page is served as a static page, therefore we need to send this event from the back-end after user activated their account
+
+How to send notification for a specific use based on their state?
+
+By utilizing customer.io’s attribute feature, we can update user’s state through each step in our referral program and deliver appropriate content to them.
+
+**How to update user’s state in customer.io?**
+
+```javascript
+analytics.identify("97980cfea0067", {
```

Comment:
Shouldn't this request be on the server side?

Reply:
Since we want to be able to track user activity on the UI, it's recommended to also call identity on all the pages that only logged in user can access in the [segment.io doc](https://segment.com/docs/spec/identify/)